server/src/main/java/org/elasticsearch/search/runtime/IpScriptFieldTermsQuery.java

```diff
@@ -16,24 +18,51 @@
 
 import java.util.Objects;
 
-public class IpScriptFieldTermsQuery extends AbstractIpScriptFieldQuery {
+import static org.elasticsearch.search.runtime.AbstractIpScriptFieldQuery.decode;
+
+public class IpScriptFieldTermsQuery extends AbstractScriptFieldQuery<IpFieldScript> {
     private final BytesRefHash terms;
 
     public IpScriptFieldTermsQuery(Script script, IpFieldScript.LeafFactory leafFactory, String fieldName, BytesRefHash terms) {
-        super(script, leafFactory, fieldName);
+        super(script, fieldName, leafFactory::newInstance);
         this.terms = terms;
     }
 
-    @Override
-    protected boolean matches(BytesRef[] values, int count) {
+    boolean matches(BytesRef[] values, int count, BytesRef intermediate) {
         for (int i = 0; i < count; i++) {
-            if (terms.find(values[i]) >= 0) {
+            if (terms.threadSafeFind(values[i], intermediate) >= 0) {
                 return true;
             }
         }
         return false;
     }
 
+    @Override
+    protected boolean matches(IpFieldScript scriptContext, int docId) {
+        throw new UnsupportedOperationException("Need to use matches(IpFieldScript, int, BytesRef) instead");
```

Comment:
I find this super ugly to do - we're extending a class, but oh wait, this method is not the right one to use 😢 . And we're no longer extending from  `AbstractIpScriptFieldQuery` like our pals, but `AbstractIpScriptFieldQuery` because otherwise we can't redefine `matches`...

I'm trying to find a better way for this, but so far I'm out of better ideas 🤷

Reply:
I think this is OK, its not so bad. The bad part is that BytesRefHash uses a friggin' piece of state for `find`, I mean, what on earth.